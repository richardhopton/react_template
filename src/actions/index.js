export const TOGGLE_MESSAGE = "TOGGLE_MESSAGE";

export const toggleMessage = value => ({
  type: TOGGLE_MESSAGE,
  value: value
});

/*
export const FETCHED_ITEMS = "FETCHED_ITEMS";

const parseResponse = json => ({
  type: FETCHED_ITEMS,
  items: json.data.children.map(child => child.data),
  lastUpdated: Date.now()
});

export const exampleFetch = () => (dispatch, getState) => {
  const url = getState().url || "something";
  return fetch(`http://www.${url}.com`)
    .then(response => response.json())
    .then(json => dispatch(parseResponse(json)));
};
*/