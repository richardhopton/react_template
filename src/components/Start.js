import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { toggleMessage } from "../actions";
import styles from "./styles.css";

export class Start extends Component {
  handleClick(e) {
    e.preventDefault();
    const { dispatch } = this.props;
    dispatch(toggleMessage());
  }

  render() {
    const { messageState } = this.props;
    return (
      <div>
        <div className={styles.header}>
          <div className={styles.headerName}>React Template</div>
        </div>
        <div className={styles.body}>
          {messageState == 0 ? "Hello World" : "Goodbye cruel world"}
          <button onClick={this.handleClick.bind(this)}>Go</button>
        </div>
      </div>
    );
  }
}

Start.propTypes = {
  messageState: PropTypes.number.isRequired,
  dispatch: PropTypes.func.isRequired
};

function mapState(state) {
  const { state: messageState } = state.message || { state: 0 };

  return {
    messageState
  };
}

export default connect(mapState)(Start);
