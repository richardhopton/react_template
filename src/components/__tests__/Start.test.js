import React from "react";
import { Start } from "../Start";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { TOGGLE_MESSAGE } from "../../actions"; 

const dispatch = jest.fn();
  
// Check the component hasn't been accidentally adjusted by using a jest snapshot
test("Start renders expected default snapshot", () => {
  const {start} = <Start messageState={0} dispatch={dispatch} />;
  const tree = renderer.create(start).toJSON();
  expect(tree).toMatchSnapshot();

});

test("Start renders expected alternate snapshot", () => {
  const {start} = <Start messageState={1} dispatch={dispatch} />;
  const tree = renderer.create(start).toJSON();
  expect(tree).toMatchSnapshot();
});

// Check interactions
test("Start toggles on button click", () => {
  const start = shallow(<Start messageState={0} dispatch={dispatch} />);

  start.find("button").simulate("click", { preventDefault: () => {} });

  expect(dispatch).toHaveBeenCalledWith({type: TOGGLE_MESSAGE});
});
