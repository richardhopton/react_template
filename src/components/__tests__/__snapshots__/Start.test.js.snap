// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start renders expected alternate snapshot 1`] = `null`;

exports[`Start renders expected default snapshot 1`] = `null`;
